### PR TITLE
Add tests and CI

### LINKED ISSUE #240: support CI for dbt-adapter-test suite
### Describe the feature

it would benefit the users to know how compliant the adapter is with the test suite provided by dbt Labs. Additionally would help prevent regressions from being introduced in future PRs.

### Additional context
Requirement for https://github.com/dbt-labs/docs.getdbt.com/issues/4091

### Who will this benefit?
dbt Labs community members who are interested in 

### Are you interested in contributing this feature?
Let me know how I can be of service!

### PR DESCRIPTION
**Pull request by moomindani**
resolves #240 

### Description

Added CI

### Checklist

- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have updated the `CHANGELOG.md` and added information about my change to the "dbt-glue next" section.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-session-name: ${{ steps.session.outputs.name }}
+          role-to-assume: arn:aws:iam::421396179189:role/dbt-glue
+          aws-region: us-east-1
+
+      - name: Run tox
+        run: |
+          export DBT_AWS_ACCOUNT=421396179189
```
Comment (nicor88) [inline .github/workflows/main.yml:122]: consider to put all AWS_ACCOUNT to secrets and fetch from there.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+          export DBT_S3_LOCATION=s3://dbt-glue-ci/data/${GITHUB_RUN_ID}-${GITHUB_RUN_ATTEMPT}/${{ matrix.python-version }}/
+          export DBT_GLUE_ROLE_ARN=arn:aws:iam::421396179189:role/GlueServiceRole
```
Comment (nicor88) [inline .github/workflows/main.yml:124]: same as https://github.com/aws-samples/dbt-glue/pull/258/files#r1386403378, put it into secrets.

**Comment by nicor88** [comment]
As a general note I recommend this approach to use in your CI:
* unit tests - where you mock as much as possible everything that you need, e.g. using moto. unit tests do not need a real account to run, and they can be run easily even locally.
* functional tests, they implement dbt-tests-adapter tests, so pretty much a real case scenario test with real models. Those tests at the contrary require a real AWS account with some infra to run on. In dbt-athena we run those conditionally in the CI (for cost reasons), but they should be separated from unit tests. I do not see any functional tests added here, and without those, you miss a big part of it.

File: .github/workflows/main.yml (hunk 1 above)
Comment (moomindani) [reply .github/workflows/main.yml]: Thanks for the feedback. Addressed.

File: .github/workflows/main.yml (hunk 2 above)
Comment (moomindani) [reply .github/workflows/main.yml]: Thanks for the feedback. Addressed.

**Comment by moomindani** [comment]
> As a general note I recommend this approach to use in your CI:
> 
>     * unit tests - where you mock as much as possible everything that you need, e.g. using moto. unit tests do not need a real account to run, and they can be run easily even locally.
> 
>     * functional tests, they implement dbt-tests-adapter tests, so pretty much a real case scenario test with real models. Those tests at the contrary require a real AWS account with some infra to run on. In dbt-athena we run those conditionally in the CI (for cost reasons), but they should be separated from unit tests. I do not see any functional tests added here, and without those, you miss a big part of it.

Thanks for the feedback. It was planned but not added yet yesterday.
Today I have separated tests into unit test and functional test, and used AWS account only for the functional test.

**Review by aajisaka — APPROVED** [review]
Looks good to me. Thank you!

**Review by menuetb — APPROVED** [review]

**Pull request merged**